src/components/Identity/createMigration.js

```diff
@@ -2,27 +2,36 @@ import { cookieJar } from "../../utils";
 import { EXPERIENCE_CLOUD_ID } from "./constants/cookieNames";
 
 export default (imsOrgId, idMigrationEnabled) => {
+  const amcvCookieName = `AMCV_${imsOrgId}`;
+
   return {
-    getEcidFromAmcvCookie(identityCookieJar) {
+    getEcidFromLegacyCookie(identityCookieJar) {
       let ecid = null;
+      const legacyItpCookieName = "s_ecid";
+
       if (idMigrationEnabled) {
-        const amcvCookieValue = cookieJar.get(`AMCV_${imsOrgId}`);
-        if (amcvCookieValue) {
+        const legacyEcidCookieValue =
+          cookieJar.get(legacyItpCookieName) || cookieJar.get(amcvCookieName);
+
+        if (legacyEcidCookieValue) {
           const reg = /(^|\|)MCMID\|(\d+)($|\|)/;
-          const matches = amcvCookieValue.match(reg);
+          const matches = legacyEcidCookieValue.match(reg);
           // Destructuring arrays breaks in IE
           // eslint-disable-next-line prefer-destructuring
-          ecid = matches[2];
-          identityCookieJar.set(EXPERIENCE_CLOUD_ID, ecid);
+          if (matches) {
```

Comment:
There are cases where the AMCV cookie might only have `version:1.2.3` by the time we read it, which will break `matches[2]`.

Reply:
Can you add this as a test case?

Reply:
Done @Aaronius, thanks.